services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -329,6 +330,22 @@ private List<AbstractVeniceService> createServices() {
         new RemoteIngestionRepairService(serverConfig.getRemoteIngestionRepairSleepInterval());
     services.add(remoteIngestionRepairService);
 
+    // HelixParticipationService below creates a Helix manager and connects asynchronously below. The listener service
+    // needs a routing data repository that relies on a connected helix manager. So we pass the listener service a
+    // future that will be completed with a routing data repository once the manager connects.
+    CompletableFuture<SafeHelixManager> managerFuture = new CompletableFuture<>();
```

Comment:
I am not sure I follow. This `CompletableFuture` doesn't create `SafeHelixManager`, does it?